src/platform_impl/windows/event_loop/runner.rs

```diff
@@ -496,57 +438,14 @@ impl<T> EventLoopRunner<T> {
-            // If we were handling events, send the MainEventsCleared message.
-            RunnerState::HandlingEvents => {
-                self.call_event_handler(Event::MainEventsCleared);
-                self.runner_state = RunnerState::HandlingRedraw;
-                self.flush_redraws();
-                self.call_event_handler(Event::RedrawEventsCleared);
-                self.runner_state = RunnerState::Idle(Instant::now());
-            }
-
+            // If we were handling redraws, send the RedrawEventsCleared message.
             RunnerState::HandlingRedraw => {
-                self.flush_redraws();
                 self.call_event_handler(Event::RedrawEventsCleared);
                 self.runner_state = RunnerState::Idle(Instant::now());
             }
 
-            // If we *weren't* handling events, we don't have to do anything.
-            RunnerState::New | RunnerState::Idle(..) => (),
-
-            // Some control flows require a NewEvents call even if no events were received. This
-            // branch handles those.
-            RunnerState::DeferredNewEvents(wait_start) => {
-                match self.control_flow {
-                    // If we had deferred a Poll, send the Poll NewEvents and MainEventsCleared.
-                    ControlFlow::Poll => {
-                        self.call_event_handler(Event::NewEvents(StartCause::Poll));
-                        self.call_event_handler(Event::MainEventsCleared);
-                        self.flush_redraws();
-                        self.call_event_handler(Event::RedrawEventsCleared);
-                    }
-                    // If we had deferred a WaitUntil and the resume time has since been reached,
-                    // send the resume notification and MainEventsCleared event.
-                    ControlFlow::WaitUntil(resume_time) => {
-                        if Instant::now() >= resume_time {
-                            self.call_event_handler(Event::NewEvents(
-                                StartCause::ResumeTimeReached {
-                                    start: wait_start,
-                                    requested_resume: resume_time,
-                                },
-                            ));
-                            self.call_event_handler(Event::MainEventsCleared);
-                            self.flush_redraws();
-                            self.call_event_handler(Event::RedrawEventsCleared);
-                        }
-                    }
-                    // If we deferred a wait and no events were received, the user doesn't have to
-                    // get an event.
-                    ControlFlow::Wait | ControlFlow::Exit => (),
-                }
-                // Mark that we've entered an idle state.
-                self.runner_state = RunnerState::Idle(wait_start)
-            }
+            // If we *weren't* handling redraws, we don't have to do anything.
+            _ => (),
```

Comment:
I'm not entirely comfortable with just doing nothing on this branch - if it's invoked, there's probably a bug higher up in the codebase caused `redraw_events_cleared` to be called at the wrong time. Not sure if we should panic or warn about this, but whichever way we go it shouldn't be silent.

Reply:
I think this handles the case were a Windows message is sent but we don't handle it at all (i.e. it does not translate to a winit event). There's a bunch of those. When one those arrives we will be woken up but we must do nothing.

Reply:
That's the case for a lot of the `RunnerState`s, but not all of them. If it's called during the `HandlingEvents` phase, this shouldn't be a no-op.

Reply:
PS: we can arrive here in `DeferredNewEvents` state which is ok and expected.
I can make the core more explicit by letting only `DeferredNewEvents` pass through and panic on other states. Or add a comment. Let me know.

Reply:
made redraw_events_cleared more explicit.

Reply:
Looks good, thanks!